app/assets/javascripts/service_types/IsServiceWorking.js

```diff
@@ -0,0 +1,186 @@
+    container: {
+      width: '100%',
+      margin: '10px auto',
+      position: 'relative'
+    },
+    chartHeight: 250,
+  },
+  a: {
+    fontSize: 24
+  },
+  buttonStyle: {
+    float: 'left',
+    margin: 5,
+  },
+  buttonSectionStyle: {
+    width: '100%',
+    height: 50
+  }
+};
+
+class IsServiceWorking extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      serviceTypeSelected: null // options: null, 502, 503
+    };
+
+    this.fetchIsServiceWorkingData = this.fetchIsServiceWorkingData.bind(this);
+    this.renderPage = this.renderPage.bind(this);
+  }
+
+  fetchIsServiceWorkingData() {
+    const {serviceTypeSelected} = this.state;
+
+    const url = `/api/is_service_working_json/${serviceTypeSelected.id}`;
+
+    return apiFetchJson(url);
+  }
+
+  render() {
+    return (
+      <div className="IsServiceWorking">
+        <ExperimentalBanner />
+        <SectionHeading>
+          "Is this Service Working for this Student?"
+        </SectionHeading>
+        <p style={{padding: '10px 0 0 10px'}}>
+          Every student is different.
+          One student might respond well to a service, while another might respond poorly.
+        </p>
+        <p style={{padding: '10px 0 10px 10px'}}>
+          Use the buttons below to explore how students may have responded to
+          services assigned to them over the past school year.
+        </p>
+        {this.renderButtons()}
+        {this.renderCharts()}
+      </div>
+    );
+  }
+
+  renderButtons() {
+    const attendanceOfficer = {
+      id: 502,
+      name: 'Attendance Officer',
+    };
+
+    const attendanceContract = {
+      id: 503,
+      name: 'Attendance Contract',
+    };
+
+    return (
+      <div style={style.buttonSectionStyle}>
+        <Button style={style.buttonStyle}
+          onClick={() => {
+            this.setState({serviceTypeSelected: null}, () => {
+              this.setState({serviceTypeSelected: attendanceOfficer});
+            });
+          }}>
+          Attendance Officer
+        </Button>
+        <Button style={style.buttonStyle}
+          onClick={() => {
+            this.setState({serviceTypeSelected: null}, () => {
+              this.setState({serviceTypeSelected: attendanceContract});
+            });
+          }}>
+          Attendance Contract
+        </Button>
+      </div>
+    );
+  }
+
+  renderCharts() {
+    const {serviceTypeSelected} = this.state;
+
+    if (!serviceTypeSelected) return null;
+
+    return (
+      <GenericLoader
+        promiseFn={this.fetchIsServiceWorkingData}
+        render={this.renderPage} />
+    );
+  }
+
+  renderPage(json) {
+    return (
+      <div style={style.container}>
+        {this.renderStudents(json)}
+      </div>
+    );
+  }
+
+  renderPhasebands(services) {
+    return services.map(service => {
+      return {
+        momentUTCstart: moment.utc(service.date_started),
+        momentUTCend: moment.utc(service.discontinued_at),
```

Comment:
not sure if null handling matters here, it's probably fine to ship without caring about that, but will some of these be null?